library/pkparse.c

```diff
@@ -1456,8 +1456,10 @@ int mbedtls_pk_parse_public_key( mbedtls_pk_context *ctx,
         if( ( pk_info = mbedtls_pk_info_from_type( MBEDTLS_PK_RSA ) ) == NULL )
             return( MBEDTLS_ERR_PK_UNKNOWN_PK_ALG );
 
-        if( ( ret = mbedtls_pk_setup( ctx, pk_info ) ) != 0 )
+        if( ( ret = mbedtls_pk_setup( ctx, pk_info ) ) != 0 ) {
+            mbedtls_pem_free( &pem );
```

Comment:
Regarding testing: the only way this leak can occur is if a memory allocation fails. We don't have a framework for causing allocations to fail, so we have no realistic way to add a non-regression test at this time.

Reply:
Our implementation uses the library on an M4 type microcontroller with just 256KB RAM, and so in our environment failing heap allocations are a distinct possibility. Your remark is quite useful.